plugins/shortcuts/src/ShortcutForm.tsx

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useEffect } from 'react';
+import { useForm, SubmitHandler, Controller } from 'react-hook-form';
+import {
+  Button,
+  CardActions,
+  CardContent,
+  makeStyles,
+  TextField,
+} from '@material-ui/core';
+import { FormValues } from './types';
+
+const useStyles = makeStyles(theme => ({
+  field: {
+    marginBottom: theme.spacing(2),
+  },
+  actionRoot: {
+    paddingLeft: theme.spacing(2),
+    paddingBottom: theme.spacing(3),
+    justifyContent: 'flex-start',
+  },
+}));
+
+type Props = {
+  formValues?: FormValues;
+  onSave: SubmitHandler<FormValues>;
+  onClose: () => void;
+};
+
+export const ShortcutForm = ({ formValues, onSave, onClose }: Props) => {
+  const classes = useStyles();
+
+  const {
+    handleSubmit,
+    reset,
+    control,
+    formState: { errors },
+  } = useForm<FormValues>({
+    mode: 'onChange',
+    defaultValues: {
+      url: formValues?.url ?? '',
+      title: formValues?.title ?? '',
+    },
+  });
+
+  useEffect(() => {
+    reset(formValues);
+  }, [reset, formValues]);
+
+  return (
+    <>
+      <CardContent>
+        <Controller
+          name="url"
+          control={control}
+          rules={{
+            required: true,
+            pattern: {
+              value: /^\//,
+              message: 'Must be a relative URL (starts with a /)',
```

Comment:
A bit sad about this, why not allow shortcuts to external URLs? 😄

Reply:
It would be _super duper cool_ to pull favicons from external sites as the shortcut icon 😮

Reply:
Yep, and we've heard this from others too! We will keep developing this plugin and will look into adding external links as well as folders sometime in the future.